### DISCUSSION TITLE
Fast Refresh stops working after a few seconds when using `next dev --turbo`

### DISCUSSION THREAD

**Discussion by brw** [vercel/next.js#68182] [Help]
### Summary

I've been trying to use Turbopack with `next dev` because with it Fast Refresh/HMR is much faster for the project I'm currently working with. But for some reason Fast Refresh always seems to stop working after a few seconds and I can't figure out why. It still works fine without Turbopack.

This happens even on a clean `create-next-app` app (all defaults) and a clean Chrome (stable) browser instance (using a temporary user data directory).

My steps:
1. Create a new app using `npm create next-app`
2. Run `npm run dev -- --turbo`
3. Load the page
4. Change something
5. Observe that it changes in the browser, and outputs `Compiled / in ..ms` to the terminal
6. Repeat 4 and 5 until after a few seconds (pretty much always within 10-15 seconds, often much sooner, sometimes even before I've made any changes at all) it just stops changing.

At this point it will also stop outputting `Compiled / in ..ms` to the terminal for changes, until I refresh the page again and the same thing happens (Fast Refresh will work but only for a few seconds). *However* Chrome devtools does continue to print `[Fast Refresh] rebuilding` on changes. So interestingly enough it does still seem to be detecting them, but nothing actually changes on the page.

My environment:
- Arch Linux (6.10.0-arch1-2)
- Nodejs v22.5.1 (also tried the latest LTS v20.16.0)
- Project is located on my root filesystem
- `fs.inotify.max_user_watches` and `fs.inotify.max_user_instances` are set sufficiently high (524288 and 8192 respectively)
- Next.js 14.2.5 (also tried 15.0.0-rc.0 and v15.0.0-canary.84)

I did not submit this as a bug report because I assume this is some issue with my environment (and it felt a bit silly to submit a default create-next-app as a minimal reproduction), but I have no idea where to look at this point.

I've been pulling my hair out over this issue for the past few hours so if anyone has any input I'd appreciate it. If you need any more info feel free to let me know.

[trace.log](https://github.com/user-attachments/files/16386679/trace.log) from `NEXT_TURBOPACK_TRACING=1 npm run dev -- --turbo` (I have no idea how to read this binary file but maybe it's useful to someone. I can see that it still gets data appended to it on every change even after it stops working).

**Comment by Ezadeen-Almoshra** [comment]
I have the same problem. Did you find a solution?

**Reply by brw** [reply]
No, not really. I just tried it again one day and somehow it worked (mostly) fine again. Fast Refresh still breaks every now and then but it seems to be way more rare. Still no idea why it happens/happened though.
